### ISSUE TITLE
Add a rust-toolchain.toml in the root directory

### ISSUE THREAD

**Issue by sfzhu93** [rust-lang/rust#119684]
Since this repo requires a nightly Rust to drive at least rust-analyzer, add a `rust-toolchain.toml` in the root directory is helpful for getting started. The content of the file is simple:
```
[toolchain]
channel = "nightly"
```
Maybe this is not a best practice, but I guess some similar approach could be helpful for getting started.

**Comment by Nilstrieb** [comment]
This repo doesn't actually use nightly, it uses beta with RUSTC_BOOTSTRAP=1 to enable unstable features. You should use the rust-analyzer config from `src/etc/rust_analyzer.json` (or something like that).

**Comment by albertlarsan68** [comment]
The easiest way to get started includes running `./x setup`, which will setup the repo to be used. It will configure rust-analyzer correctly, to use the tools necessary to work with rustc or the standard library. Using "bare" rust-analyzer is discouraged, as it may mess with your ability to work efficiently.

**Comment by sfzhu93** [comment]
@albertlarsan68 @Nilstrieb Thank you for the explanation! I believe I followed the official setup steps and copied `rust_analyzer.json`. Maybe it is misconfigured on my side, but my rust-analyzer did tell me it had to use nightly version in one directory. I also remembered at least part of the code base requires nightly Rust from rust-dev-guide. 
Also, the official `rust_analyzer.json` did not specify which rust version channel should use. 
I will try to reproduce the problem again and figure out some more details.
